### PR TITLE
10 get api articles queries

### PR DESCRIPTION
**Pull request by bergamotBen**
task 10 complete and all tests passing

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -81,6 +81,36 @@ describe("GET /api/articles", () => {
...
+  it("returns 200 and articles with a topic of 'mitch'", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body }) => {
+        body.articles.map((article) => {
+          expect(article).toHaveProperty("topic");
+          expect(article.topic).toBe("mitch");
+        });
+      });
```
Comment (NRMSMITH) [inline __tests__/app.test.js:95]: a forEach would be better suited here, just make sure to check the length before. A map produces a new array, which shouldn't really be happening in tests.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -81,6 +82,52 @@ describe("GET /api/articles", () => {
...
+  it("returns 400 and rejects an invalid order", () => {
+    return request(app)
+      .get("/api/articles?order=order")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("invalid type");
+      });
+  });
+});
+
```
Comment (NRMSMITH) [inline __tests__/app.test.js:138]: A couple of tests to consider: 200 - valid topic but no articles / 404 - non-existent topic

File: models/models.js [hunk 3]
```diff
@@ -6,35 +6,52 @@ const readTopics = () => {
...
-      FROM articles
-      LEFT JOIN comments on comments.article_id = articles.article_id
-      GROUP BY articles.article_id
-      ORDER BY articles.created_at DESC;`
-    )
-    .then((articles) => {
-      return articles.rows;
-    });
-};
+const readArticles = (query) => {
```
Comment (NRMSMITH) [inline models/models.js:9]: Would you consider passing the sort_by and the order through the parameters? Considering that they are variables that change dependent on the input of the client?

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
Just a couple of things to consider - some testing and making your code in model a bit drier.

**Review by Rosem10 — APPROVED** [review]

**Pull request merged**